src/models/AbstractEntity.php

```diff
@@ -786,4 +778,13 @@ private function getReadSqlBeforeWhere(bool $getTags = true, bool $fullSelect =
         // replace all %1$s by 'experiments' or 'items'
         return sprintf(implode(' ', $sqlArr), $this->type);
     }
+
+    private function bindFilterValues(array $bindValues, PDOStatement $req): void
+    {
+        if (!empty($bindValues)) {
```

Comment:
you don't need to check if it's not empty, a foreach on empty array is perfectly fine (also you missed a closing parenthesis (now fixed)).

You should install the pre-commit hook script to avoid such issues (missing parenthesis) ;)

Reply:
forgot the link: https://doc.elabftw.net/contributing.html#install-a-pre-commit-hook